components/raftstore/src/store/worker/cleanup_sst.rs

```diff
@@ -64,12 +67,31 @@ where
         }
     }
 
+    fn get_region_by_meta(&self, sst: &SstMeta) -> Result<Region> {
+        // The SST meta has been delivered with a range, use it directly.
+        if !sst.get_range().get_start().is_empty() || !sst.get_range().get_end().is_empty() {
+            return self
+                .pd_client
+                .get_region(sst.get_range().get_start())
```

Comment:
can we read the kv range from rocksdb SST directly?

Reply:
I will give it a try... But I'm not sure whether time is enough.